todo-mvc-kitchensink/tests/support/ClientErrorCollector.ts

```diff
@@ -0,0 +1,115 @@
+const CLIENT_ERROR_STACK_KEY = '__intern_error_stack';
+const CLIENT_OLD_ONERROR_KEY = '__intern_old_onerror';
+const CLIENT_FINISH_FUNCTION_KEY = '__intern_error_helper_finish';
+
+export interface ClientError {
+	message: string;
+	type: string;
+	source: string;
+	lineno: number;
+	colno: number;
+	error: {
+		message: string;
+		name: string;
+		stack?: string;
+	};
+}
+
+export default class ClientErrorCollector {
+	private _remote: any;
+	private _inited: boolean = false;
+
+	constructor (remote: any) {
+		this._remote = remote;
+	}
+
+	assertNoErrors(message?: string): Promise<undefined> {
+		return this.finish()
+			.then((results) => {
+				if (results && results.length) {
+					const result = results[0];
+					let e: Error;
+					switch (result.type) {
+					case 'TypeError':
+						e = new TypeError(result.message);
+						break;
+					case 'SyntaxError':
+						e = new SyntaxError(result.message);
+						break;
+					default:
+						e = new Error(result.message);
+					}
+					e.stack = result.error.stack;
+					(<any> e).fileName = result.source;
+					(<any> e).lineNumber = result.lineno;
+					(<any> e).columnNumber = result.colno;
+					throw e;
+				}
+			});
+	}
+
+	finish(): Promise<ClientError[] | undefined> {
+		if (!this._inited) {
+			throw new Error('ClientErrorCollector not initialised.');
+		}
+
+		return this._remote
+			.execute(`return window.${CLIENT_FINISH_FUNCTION_KEY}();`)
+			.then((results: string | undefined) => {
+				if (results) {
+					return JSON.parse(results);
+				}
+			});
+	}
+
+		init(): Promise<void> {
+		if (this._inited) {
+			throw new Error('ClientErrorCollector already initialised.');
+		}
+
+		return this._remote
+			.execute(`
+				var errorStack = window.${CLIENT_ERROR_STACK_KEY} = [];
```

Comment:
Really not a fan of code as a string. You can pass a function and args to `.execute`. Making this a real function will catch any syntax errors, allow you to type it, and not have to write es5 code.